src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::error::Failed;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+pub trait NBDistribution {
+    fn fit<T: RealNumber, M: Matrix<T>>(&self, x: &M, y: M::RowVector) -> Self;
+
+    fn prior<T: RealNumber>(k: T) -> T;
+
+    fn conditional_probability<T: RealNumber, M: Matrix<T>>(k: T, j: M::RowVector) -> T;
+}
+
+pub struct BaseNaiveBayes;
+
+impl BaseNaiveBayes {
+    pub fn fit<D: NBDistribution>(&mut self, distribution: D) -> Result<(), Failed> {
```

Comment:
Shell we return an instance of `BaseNaiveBayes` here? Otherwise, what user will get after fitting `BaseNaiveBayes ` to data? Also, method `fit` should be static.
```suggestion
    pub fn fit<D: NBDistribution>(distribution: D) -> Result<Self, Failed> {
```